src/main/java/com/faforever/client/remote/FafService.java

```diff
@@ -458,13 +459,27 @@ public void restoreGameSession(int id) {
   }
 
   @Async
-  public CompletableFuture<Tuple<List<MapBean>, Integer>> getLadder1v1MapsWithPageCount(int count, int page) {
-    Tuple<List<Ladder1v1Map>, java.util.Map<String, ?>> tuple = fafApiAccessor.getLadder1v1MapsWithMeta(count, page);
-    return CompletableFuture.completedFuture(new Tuple<>(tuple.getFirst()
-        .parallelStream()
-        .map(ladder1v1Map -> MapBean.fromMapVersionDto(ladder1v1Map.getMapVersion()))
+  public CompletableFuture<Tuple<List<MapBean>, Integer>> getMatchmakerMapsWithPageCount(int matchmakerQueueId, int count, int page) {
+    List<MapVersion> mapVersions = fafApiAccessor.getMatchmakerPools(matchmakerQueueId)
+        .stream()
+        .map(MatchmakerQueueMapPool::getMapPool)
+        .flatMap(mapPool -> mapPool.getMapVersions().stream())
+        .distinct()
+        .collect(toList());
+    int totalPages = (mapVersions.size() - 1) / count + 1;
+    return CompletableFuture.completedFuture(new Tuple<>(mapVersions
+        .stream()
+        .skip((page - 1) * count)
+        .limit(count)
+        .map(MapBean::fromMapVersionDto)
         .collect(toList()),
-        ((HashMap<String,Integer>) tuple.getSecond().get("page")).get("totalPages")));
+        totalPages));
+  }
+
```

Comment:
What exactly? Or the whole thing?

Reply:
The integer that is returned as well. Calculating page sizes / page numbers is not a problem tied to map beans, it applies to every collection and it could go into some util function.

Reply:
Just as a note for all other api calls the api gives us the total page number. However it doesnt do that for the mapPool maps currently which is why it needs to be calculated here.

Reply:
I agree, but all the other api calls use pagination to limit the api call itself. This one is the only one (yet) that has to limit the returning result. What would an appropriate place for the helper function be?